spec.bs

```diff
@@ -134,33 +192,99 @@ Modify {{Document/requestStorageAccess()}} at step 14.1.1.1.1 to read:
 
 <h3 id="storage">Changes to various client-side storage mechanisms</h3>
 
-<h4 id="dom-storage">DOM Storage</h4>
+For all of the following getters and methods, consider the |docForSAA| to be |doc| with the following modifications:
 
-TBD
+1. When attempting to [=obtain a storage key=] the returned key will use [[STORAGE-PARTITIONING#relaxing-additional-keying]] if the tuple does not simply contain an [=/origin=].
 
-<h4 id="indexed-db">IndexedDB</h4>
+Issue(19): Clarify client-side storage mechanism changes in more detail.
 
-TBD
+<h4 id="dom-storage">[=Web storage=]</h4>
 
-<h4 id="web-locks">Web Locks</h4>
+When invoked on {{StorageAccessHandle}} |handle| with {{StorageAccessTypes}} |types| and {{Document}} |docForSAA|, the <dfn export attribute for=StorageAccessHandle><code>sessionStorage</code></dfn> getter must run these steps:
 
-TBD
+1. If |types|.{{StorageAccessTypes/all}} is `false` and |types|.{{StorageAccessTypes/sessionStorage}} is `false`:
+    1. Throw an "{{InvalidStateError}}" {{DOMException}}.
+1. Return the invocation of [=html/sessionStorage=] on |docForSAA|.
```

Comment:
As mentioned above, sessionStorage is not actually invoked on a document, it gets the window's (this) associated document. But I don't think that actually matters so much, the handle holds the information that is sufficient to tell the getter that it needs a different storage key.

So I think we can remove the mention of |docForSAA| here, WDYT?